text/0000-newtype.md

```diff
@@ -0,0 +1,360 @@
+
+Let `R1` be the binary, transitive, symmetric, irreflexive relation generated by
+all `newtype` definitions, all `alias` definitions, and the numeric casts built
+into the language. 
+
+If `(T, U) \in R1` and `x: T`, then `x as U` is a valid cast.
+
+Let `R2` be the binary, transitive, symmetric, irreflexive relation generated by
+all `newtype` definitions and all `alias` definitions.
+
+Let `S` be the set of reference operators, `S = {&, &mut}`, and let `W` be a
+word generated by `S`, e.g., `W = e` or `W = &&&mut &&mut &`.
+
+If `(T, U) \in R2` and `x: WT`, then `x as WU` is a valid cast.
+
+### Inference
+
+Let `R2` be as above.
+
+If `U` is an integer type and `(T, U) \in R2`, then untyped integer literals will
+be infered as `T` in `T` position.
+
+If `U` is a floating point type and `(T, U) \in R2`, then untyped floating point
+literals will be infered as `T` in `T` position.
+
+### Traits
+
+No built-in traits are or can be implemented for `T` except as described below.
+
+For all types `W`, `PartialOrd<W>` can be implemented for `T`. If
+`PartialOrd<T>` is not explicitly implemented for `T` and `PartialOrd<U>` is
+implemented for `U`, then `PartialOrd<T>` shall automatically be implemented for
+`T` via casting to `U`.
+
+`Neg` can be implemented for `T`. If `Neg` is not explicitly implemented for `T`
+and `Neg` is implemented for `U`, then `Neg` shall automatically be implemented
+for `T` via casting to `U`.
+
+For all types `W`, `Shl<W>` can be implemented for `T`. If for any integer type
+`I` `Shl<I>` is not is not explicitly implemented for `T` and `Shl<I>` is
+implemented for `U`, then `Shl<I>` shall automatically be implemented for `T`
+via casting to `U`.
+
+`T` implements `Drop` if and only if `U` implements `Drop`.
+
+`T` implements `Send` if and only if `U` implements `Send`.
+
+`T` implements `Sized` if and only if `U` implements `Sized`.
+
+`T` implements `Copy` if and only if `U` implements `Copy`.
+
+`T` implements `Sync` if and only if `U` implements `Sync`.
+
+For all types `W`, `Fn<W>` can be implemented for `T`.
+
+For all types `W`, `Index<W>` can be implemented for `T`.
+
+For all types `W`, `IndexMut<W>` can be implemented for `T`.
+
+`Deref` can be implemented for `T`.
+
+`DerefMut` can be implemented for `T`.
+
+The same rules as for `PartialOrd` apply to `PartialEq`.
+
+The same rules as for `PartialOrd` apply to `Add`.
+
+The same rules as for `PartialOrd` apply to `Sub`.
+
+The same rules as for `PartialOrd` apply to `Mul`.
+
+The same rules as for `PartialOrd` apply to `Div`.
+
+The same rules as for `PartialOrd` apply to `Rem`.
+
+The same rules as for `Neg` apply to `Not`.
+
+The same rules as for `PartialOrd` apply to `BitAnd`.
+
+The same rules as for `PartialOrd` apply to `BitOr`.
+
+The same rules as for `PartialOrd` apply to `BitXor`.
+
+The same rules as for `Shl` apply to `Shr`.
+
+The same rules as for `Fn` apply to `FnMut`.
+
+The same rules as for `Fn` apply to `FnOnce`.
+
+#### Default impls
```

Comment:
This rule is also implied by the OIBIT rules, i.e. could be written as "Traits with default impls behave as if `newtype T = U;` is the same as `struct T(U);`."